src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java

```diff
@@ -74,25 +79,27 @@ private CloudWatchCollector(Map<String, Object> config, AmazonCloudWatchClient c
         loadConfig(config, client);
     }
 
-    private ArrayList<MetricRule> getRules() {
-        synchronized (rules) {
-            return rules;
+    private AmazonCloudWatchClient getClient() {
```

Comment:
This is still racey. Both need to be fetched at the same time for a given scrape.

Reply:
Made some changes in 9705710 to clone the active config that's in memory.